fix style in app as per rubocop

georgestam opened the pull request:
@vemv I have added `rubocop` and fixed almost everything. Then, I fixed a number of things also but I think that the following we would leave it for now... as it is too much work (not sure about the last one):

![screen shot 2017-03-26 at 19 58 47](https://cloud.githubusercontent.com/assets/16335156/24334236/01acd7ac-125f-11e7-8801-233334e47e7d.png)

vemv commented on spec/factories/imparted_hour_factory.rb:
```diff
@@ -5,7 +5,7 @@
     default_minutes = Random.rand(1..120)
     minutes { default_minutes }
-    date { Date.today }
+    date { Time.zone.today }
```
use `Time.current.to_date` instead. Same result, but you should be using `Time.current` for almost everything.

The reason is that 1 thing is easier to memorize than N things

vemv commented on app/controllers/api/v1/base_controller.rb:
```diff
@@ -1,31 +0,0 @@
-class Api::V1::BaseController < ActionController::Base
```
did rubocop detect an unused controller?

georgestam replied:
Rubocop detected a number of offences, as it is not used at all, I decided to remove it. 
![screen shot 2017-03-27 at 09 19 57](https://cloud.githubusercontent.com/assets/16335156/24346966/98c2d668-12ce-11e7-91f0-fbf658387455.png)

vemv replied:
that kind of action deserves its own commit :)

georgestam replied:
Ok, I put back the code and will remove it in another commit.

vemv commented on app/controllers/profiles_controller.rb:
```diff
@@ -33,12 +32,12 @@ def dashboard
     @user = current_user
     authorize @user
     @teacher = false
+    car = @user.cars.first ? @user.cars.first : nil
```
leave it as `car = @user.cars.first`

vemv commented on app/controllers/profiles_controller.rb:
```diff
@@ -116,7 +114,7 @@ def find_car
   def user_params
-    params.require(:user).permit(:photo, :id_document, :first_name, :last_name, :email, :phone_number, :description, :gender, :facebook_URL, :linkedin_URL, :address, :date_of_birth)
+    params.require(:user).permit(:photo, :id_document, :first_name, :last_name, :email, :phone_number, :description, :gender, :facebook_url, :linkedin_url, :address, :date_of_birth)
   end
```
not sure if params are case-insensitive. I recommend downcase as rubocops does, but in that case ensure that this change is applied everywhere

vemv commented on app/views/cars/_card.html.erb:
```diff
@@ -93,7 +93,7 @@
     <li>
       <p>
-        <% if car.user.facebook_URL != "" %>
+        <% if car.user.facebook_url != "" %>
```
careful with this case change, as said

vemv commented on config/database.yml:
```diff
@@ -67,15 +67,15 @@ test:
 #
 #   production:
-#     url: <%= ENV['DATABASE_URL'] %>
+#     url: <%= ENV['DATABASE_url'] %>
```
I'd just remove the rubocop rule which is rewriting "URL". Seems too dangerous.

Use `bundle exec rubocop -d` to determine which rule is being applied, then add it to rubocop.yml

georgestam replied:
The whole things about `URL` upcase/downcase was my fault (I did a 'Replace All' destruction). Now I have only changed the instance variables that `robocop complained, which were less than 10.

vemv commented on config/environments/development.rb:
```diff
@@ -15,7 +15,7 @@
 
   # Enable/disable caching. By default caching is disabled.
-  if Rails.root.join('tmp/caching-dev.txt').exist?
+  if Rails.root.join('path', 'to').exist?
```
looks strange.

georgestam replied:
![screen shot 2017-03-27 at 10 18 10](https://cloud.githubusercontent.com/assets/16335156/24349376/c813fcd2-12d6-11e7-9629-e9b854faccb3.png)

vemv replied:
it's complaining about the slash. Prefer `.join` to `/`. "path" and "to" are examples.

vemv replied:
not fixed

georgestam replied:
`Rails.root.join('tmp/caching-dev.txt').exist`is a default comand from rails 5. I changed it now but I am unsure it is correct (http://stackoverflow.com/questions/3724487/rails-root-directory-path)

vemv replied:
it's correct now. Anyway you could have tried both pieces in `rails c` and checking that they are in fact equivalent

vemv commented on config/initializers/devise.rb:
```diff
@@ -249,8 +249,8 @@
-  Devise.setup do |config|
-    config.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
+  Devise.setup do |con|
+    con.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
```
bizarre variable name change! was it automatic? if so, avoid it with rubocop.yml (as said above)

georgestam replied:
Rubocop complained about the use of the use of word `config`, that's why I changed to `con`. I changed now to `var`? (by the way, the feature with facebook login it is disabled for now)

![screen shot 2017-03-27 at 10 24 44](https://cloud.githubusercontent.com/assets/16335156/24349660/a9333fca-12d7-11e7-9bc8-2a3bb25bca4c.png)

vemv replied:
You're fixing the symptom rather than the cause. Nesting `Devise.setup` blocks one inside the other makes no sense. Extract the content of the inner one, and put it in the outer one.

georgestam commented:
@vemv I have implemented the changes as follows:
- Commit bdd9539 shows all the changes that `rubocop` fixed automatically
- Commit 4efa3ec shows the changes that I fixed manually. There are still 6 offences as per the first snapshot above. 

(Commit 	eb834d4 fix some issues with pundit that I did not identify earlier... due to the lack of testing!)

vemv commented on config/initializers/devise.rb:
```diff
@@ -249,8 +249,8 @@
-  Devise.setup do |config|
-    config.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
+  Devise.setup do |var|
+    var.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
```
not fixed, avoid nesting Devise.setup inside Devise.setup

georgestam replied:
Not sure what I have to do here. I would need to study properly how omniauth works (I was not the one who wrote this code). I would propose to remove everything related with omniauth and implement this later in a new commit.

vemv replied:
> avoid nesting Devise.setup inside Devise.setup
do this, nothing else needed

vemv commented on spec/factories/imparted_hour_factory.rb:
```diff
@@ -5,7 +5,7 @@
     default_minutes = Random.rand(1..120)
     minutes { default_minutes }
-    date { Date.today }
+    date { Time.current }
```
`Time.current.to_date`? (I assume we want a Date object)

vemv commented on app/controllers/profiles_controller.rb:
```diff
@@ -57,24 +56,20 @@ def dashboard
 
       @passengers = Passenger.where(journey_id: Journey.where(car: @car))
-
-      end
```
This moved `end` (it's below now) might change meaning. Or maybe git generated a strange diff (happens sometimes)?

georgestam replied:
Yes, it is correct. The code worked also before but now is better, as a part of the code is not run now if the teacher is the current user.

georgestam commented:
@vemv I have implemented the small change

vemv commented on db/seeds.rb:
```diff
@@ -165,27 +156,26 @@
-  admin.save
-
+admin.photo = open("http://res.cloudinary.com/georgestam/image/upload/v1476103177/dtwp5e5fj2vqiv0mkacn.jpg")
+admin.save
```
.save!
